### PR TITLE
feat: add 1inch donations

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Adds opt-out donations to 1inch.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

Small

## Testing

Ensure `REACT_APP_FEATURE_ONE_INCH` is enabled.

Test that donations work when using the 1inch swapper. Donated amounts should be received by the DAO treasury.

### Engineering

☝️

### Operations

☝️

## Screenshots (if applicable)

Screenshot 2023-05-05 at 2 53 42 pm

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4434** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4434" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4434?utm_source=stack-comment).

File: src/state/zustand/swapperStore/selectors.ts [hunk 1]
```diff
@@ -95,11 +95,11 @@ export const selectSwapperDefaultAffiliateBps = (state: SwapperState): string =>
   switch (swapperName) {
     case SwapperName.Thorchain:
     case SwapperName.Zrx:
+    case SwapperName.OneInch:
```
Comment (woodenfurniture) [inline src/state/zustand/swapperStore/selectors.ts:98]: 🚀

File: src/components/Trade/TradeConfirm/TradeConfirm.tsx [hunk 2]
```diff
@@ -433,10 +433,11 @@ export const TradeConfirm = () => {
 
   const isTHORChainSwap = useMemo(() => swapperName === SwapperName.Thorchain, [swapperName])
   const is0xSwap = useMemo(() => swapperName === SwapperName.Zrx, [swapperName])
+  const is1inchSwap = useMemo(() => swapperName === SwapperName.OneInch, [swapperName])
 
   const shouldShowDonationOption = useMemo(() => {
-    return (isTHORChainSwap || is0xSwap) && !isDonationAmountBelowMinimum
-  }, [is0xSwap, isDonationAmountBelowMinimum, isTHORChainSwap])
+    return (isTHORChainSwap || is0xSwap || is1inchSwap) && !isDonationAmountBelowMinimum
+  }, [is0xSwap, isDonationAmountBelowMinimum, isTHORChainSwap, is1inchSwap])
```
Comment (woodenfurniture) [inline src/components/Trade/TradeConfirm/TradeConfirm.tsx:440]: github wont let me put a suggestion here but maybe we can do the following?

```
  const shouldShowDonationOption = useMemo(() => {
    return [].includes(swapperName) && !isDonationAmountBelowMinimum
  }, [swapperName, isDonationAmountBelowMinimum])
```

this way we're avoiding 5 additional hash operations per render :)

**Review by woodenfurniture — DISMISSED** [review]
one non-blocking comment 🚢

**Pull request merged**